### PR TITLE
Simplify strError tests to avoid differencies between platforms

### LINKED ISSUE #763: False Negatives in Unit Tests
The following are cosmetic.  However, it's probably a small effort to hide these distractions.

1) On Cygwin and MinGW:
```
[----------] 3 tests from strError
[ RUN      ] strError.returnSuccessAfterClosingFile
/home/rmills/gnu/github/exiv2/buildserver/unitTests/test_futils.cpp:31: Failure
      Expected: expectedString
      Which is: "Success (errno = 0)"
To be equal to: strError().c_str()
      Which is: "No error (errno = 0)"
[  FAILED  ] strError.returnSuccessAfterClosingFile (2 ms)
[ RUN      ] strError.returnNoSuchFileOrDirectoryWhenTryingToOpenNonExistingFile
[       OK ] strError.returnNoSuchFileOrDirectoryWhenTryingToOpenNonExistingFile (0 ms)
[ RUN      ] strError.doNotRecognizeUnknownError
[       OK ] strError.doNotRecognizeUnknownError (0 ms)
[----------] 3 tests from strError (2 ms total)
```
I'm not sure of the origin of strings 'Success' or 'No Error', however we can evade this by only report `if error != 0`.

2) On all platforms, we're getting date/time warnings, from tests which correctly warn as follows:
```Warning: Unsupported time format```

For example (on Linux)
```
[----------] 9 tests from ADateValue
[ RUN      ] ADateValue.isDefaultConstructed
[       OK ] ADateValue.isDefaultConstructed (0 ms)
...
[ RUN      ] ADateValue.doNotReadFromByteBufferWithoutExpectedSize
Warning: Unsupported date format
....
```

The flag `--quiet` on the exiv2 command-line program will suppress warnings and we should consider using that on those tests.

### LINKED ISSUE #803: Install GMock in the Gitlab jobs missing it
In #802 we have some builds failing because I stared using GMock there but it is not installed in some of the Gitlab distributions.

### PR DESCRIPTION
**Pull request by piponazo**
This PR fixes #763. This PR fixes #803.

I simplified the tests for `strError` so that we do not need to write complex if-else preprocessor conditions to check the slightly different messages in every platform.

### CODE REVIEW

**Comment by D4N** [comment]
Lgtm, but gitlab is unhappy

**Comment by piponazo** [comment]
Damn it ... gmock is not available in some of the Gitlab distros. I'll work to bring it. I expect to use GMock to mock some classes in the future.

**Comment by clanmills** [comment]
You may (or may not) find my notes helpful:  https://github.com/Exiv2/exiv2/issues/575

**Comment by D4N** [comment]
I've tried adding gmock in via 38b8ad7, let's see whether that works.

File: ci/install_dependencies.sh [hunk 1]
```diff
@@ -24,23 +24,24 @@ distro_id=$(grep '^ID=' /etc/os-release|awk -F = '{print $2}'|sed 's/\"//g')
...
     'fedora')
-        dnf -y --refresh install gcc-c++ clang cmake make ccache expat-devel zlib-devel libssh-devel libcurl-devel gtest-devel which dos2unix
+        dnf -y --refresh install gcc-c++ clang cmake make ccache expat-devel zlib-devel libssh-devel libcurl-devel gtest-devel gmock-devel which dos2unix
         ;;
 
     'debian')
+        echo "deb http://deb.debian.org/debian stretch-backports main contrib non-free" >> /etc/apt/sources.list
         apt-get update
-        apt-get install -y cmake g++ clang make ccache python3 libexpat1-dev zlib1g-dev libssh-dev libcurl4-openssl-dev libgtest-dev libxml2-utils
+        apt-get install -y cmake g++ clang make ccache python3 libexpat1-dev zlib1g-dev libssh-dev libcurl4-openssl-dev libgtest-dev libgmock-dev libxml2-utils
```
Comment (nehaljwani) [inline ci/install_dependencies.sh]:
```suggestion
        apt-get install -y cmake g++ clang make ccache python3 libexpat1-dev zlib1g-dev libssh-dev libcurl4-openssl-dev googletest libxml2-utils
```
The new package name is just: googletest
```
root@7fbc3ab1d982:/# cat /etc/issue 
Debian GNU/Linux 9 \n \l

root@7fbc3ab1d982:/# apt-file search gmock/gmock.h
googletest: /usr/include/gmock/gmock.h
googletest: /usr/src/googletest/googlemock/include/gmock/gmock.h

root@7fbc3ab1d982:/# apt-file search gtest/gtest.h
googletest: /usr/include/gtest/gtest.h
googletest: /usr/src/googletest/googletest/include/gtest/gtest.h
```

File: ci/install_dependencies.sh [hunk 2]
```diff
@@ -24,23 +24,24 @@ distro_id=$(grep '^ID=' /etc/os-release|awk -F = '{print $2}'|sed 's/\"//g')
 
 case "$distro_id" in
     'fedora')
-        dnf -y --refresh install gcc-c++ clang cmake make ccache expat-devel zlib-devel libssh-devel libcurl-devel gtest-devel which dos2unix
+        dnf -y --refresh install gcc-c++ clang cmake make ccache expat-devel zlib-devel libssh-devel libcurl-devel gtest-devel gmock-devel which dos2unix
         ;;
 
     'debian')
+        echo "deb http://deb.debian.org/debian stretch-backports main contrib non-free" >> /etc/apt/sources.list
```
Comment (nehaljwani) [inline ci/install_dependencies.sh]:
```suggestion
```

**Comment by nehaljwani** [comment]
CI is happy now!

**Comment by piponazo** [comment]
Now, I only need an approval! 😇

**Review by D4N — CHANGES_REQUESTED** [review]
I'll cleanup the commit log, then I'll approve.

**Review by D4N — APPROVED** [review]

**Pull request merged**